r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/Capability.java

```diff
@@ -169,7 +169,11 @@ public final class Capability {
 //    private static final long MARIADB_CLIENT_PROGRESS = 1L << 32;
 //    private static final long MARIADB_CLIENT_COM_MULTI = 1L << 33;
 //    private static final long MARIADB_CLIENT_STMT_BULK_OPERATIONS = 1L << 34;
-//    private static final long MARIADB_CLIENT_EXTENDED_TYPE_INFO = 1L << 35;
+
+    /**
+     * Receive extended column type information from MariaDB to find out more specific details about column type.
+     */
+    private static final long MARIADB_CLIENT_EXTENDED_TYPE_INFO = 1L << 35;
 //    private static final long MARIADB_CLIENT_CACHE_METADATA = 1L << 36;
 
     private static final long ALL_SUPPORTED = CLIENT_MYSQL | FOUND_ROWS | LONG_FLAG | CONNECT_WITH_DB |
```

Comment:
When I add the MARIADB_CLIENT_EXTENDED_TYPE_INFO capability to ALL_SUPPORTED, I get an error. Could you check this out?

Reply:
```java
private static final long ALL_SUPPORTED = CLIENT_MYSQL | FOUND_ROWS | LONG_FLAG | CONNECT_WITH_DB |
                                              NO_SCHEMA | COMPRESS | LOCAL_FILES | IGNORE_SPACE | PROTOCOL_41 |
                                              INTERACTIVE | SSL |
                                              TRANSACTIONS | SECURE_SALT | MULTI_STATEMENTS | MULTI_RESULTS |
                                              PS_MULTI_RESULTS |
                                              PLUGIN_AUTH | CONNECT_ATTRS | VAR_INT_SIZED_AUTH | SESSION_TRACK |
                                              DEPRECATE_EOF | ZSTD_COMPRESS | MARIADB_CLIENT_EXTENDED_TYPE_INFO
```

Reply:
When I was testing it the MariaDB versions >= 10.5 didn't seem to return the extended_type_info capability which is what confused me as I manually enabled these capabilities in the test and it didn't work because the packet didn't contain the extended information. In this case the capability is enabled by default but the packet doesn't contain the extended information so if anything you shouldn't include the extended_type_info capability in ALL_SUPPORTED and instead rely on the server returning the capability if enabled. But then the capability should have been turned off if not enabled by the server should it not..

Reply:
Oh, I understand. In my setup, [the capability variable](https://github.com/asyncer-io/r2dbc-mysql/blob/acff42903b2a54a8d526f4ec2c7fc38c397fc567/r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/message/server/HandshakeV10Request.java#L138-L138) represents the server-side capability, which is provided directly by the server. This capability includes extended_type_info, but the capability.extendMariaDb line [here](https://github.com/asyncer-io/r2dbc-mysql/blob/acff42903b2a54a8d526f4ec2c7fc38c397fc567/r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/message/server/HandshakeV10Request.java#L145-L145) removes this flag due to the influence of [ALL_SUPPORTED](https://github.com/asyncer-io/r2dbc-mysql/blob/acff42903b2a54a8d526f4ec2c7fc38c397fc567/r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/Capability.java#L380-L380).

Reply:
@svats0001 Could you try using MariaDB version 11.5.2? Alternatively, you could run the test with the VM options `-Dtest.db.type=mariadb -Dtest.db.version=11.5.2`. The official Docker image supports the flag by default.

Reply:
Oh I overlooked that fact about ALL_SUPPORTED, that makes sense.

I've tried with version 11.5.2 as well and it's not working either. The fact that all the tests run fine without trying to read extended_type_info suggests that the column definition packets don't contain this information. In the docs for extended_type_info, it does say 'while string has data' which may mean that it only returns the data for specific columns, so I'll play around with that some more by testing conditionally on the value of the first byte. Also, even versions < 10.5 return the extended_type_info flag because the tests fail for these versions too which is weird.

Reply:
It looks like I'm not able to progress this issue any further, I'll close this pull request if that's okay @jchrys

Reply:
@svats0001 
Would it be alright to keep this PR open so I can continue the work when I have time? I believe it deserves to be completed and delivered!

Reply:
Yep no problem, I'll leave it open.